src/hotspot/share/gc/shenandoah/heuristics/shenandoahYoungHeuristics.cpp

```diff
@@ -150,8 +149,11 @@ bool ShenandoahYoungHeuristics::should_start_gc() {
 
   ShenandoahOldHeuristics* old_heuristics = heap->old_heuristics();
   size_t mixed_candidates = old_heuristics->unprocessed_old_collection_candidates();
-  if (mixed_candidates > 0) {
+  if (mixed_candidates > ShenandoahExpediteMixedThreshold && !heap->is_concurrent_weak_root_in_progress()) {
```

Comment:
Might be a good idea to move the counting & test of threshold inside the test for collection state.